### ISSUE TITLE
Adult mod Zaz Animation Pack does not work

### ISSUE THREAD

**Issue by TheMud** [Monitor144hz/Pandora-Behaviour-Engine-Plus#56]
All animations in ZAP don't seem to work. I see it in the list of loaded mods when launching Pandora, but neither equipping restraining items nor trying to use bondage furniture does anything.

**Comment by NefYmind** [comment]
Just wanted to make sure you were aware of this - https://www.loverslab.com/topic/100381-zaz-animation-packs-for-se/?do=findComment&comment=4260349

**Comment by AntimonyF5** [comment]
And mod animations that require ZAZ as a pre-requirement don't work. For example pama prison alternative.

**Comment by 7Costake7** [comment]
i just tryed this ,,  https://www.loverslab.com/topic/100381-zaz-animation-packs-for-se/?do=findComment&comment=4260349 ,, it didnt work for me (the character stays idle near the furniture)

**Comment by shane400m** [comment]
same for me, animations dont play and the PC stays idled next to the furniture. Its supposedly "working" as the few seconds to exit/play dismount animation still happen before movement controls are regained.

**Comment by Monitor144hz** [comment]
I suspect this to be a dupe of #76 , as furniture behaviours are probably also supposed to be handled by the engine and not packed within the FNIS file

**Comment by VredemanH** [comment]
for me, all humans stays in "A" pose with zaz enable... trying diferent things but i guess we will have to wait for updates on pandora (just saying, amazing work you guys are doing thank you!)

**Comment by anonSkyrim95** [comment]
Hello, I am fascinated by Pandora and the speed, but Zaz does not work, although not entirely true, for those who have Zaz and use Pandora can go to the ZAZ MCM and choose a character with the N button and go to the slots tab select the actor, in items add for example a yoke and you will notice that in the bound hands animations part a name that starts with zbf (something), change it to just yoke and in forced animation select any one, in this way the animation of zaz plays. but that's only for characters, not furniture. I'm thinking that for some reason the animations or furniture are written with () or begin with another name zbf in their name and this may affect the incorporation of Pandora. I'll keep trying because I'm really trying everything. I hope it works because even if I don't want to, Zaz is a bital mod for mods in which I like to save people all over Skyrim inquisitorial style.

**Comment by anonSkyrim95** [comment]
I did it, I MADE ZAZ WORK IN PANDORA HAHAHAHAHAHA, I will put together a tutorial but in simple words when generating the animations with Pandora the defaultmale.hkx and defaultfemale.hkx files in the \meshes\actors\character folder are not able to correctly group the number of animations from zaz 8+ therefore I generate with all the animations using FNIS those elements are clean, then I generate the pandora in another folder, and I replace that data with what was generated in pandora, good for the zap furniture animations and everything else It will work without the problem of nemesis animations. It took me 7 days but I did it. I still have to do more tests but it worked.

**Comment by VredemanH** [comment]
> I did it, I MADE ZAZ WORK IN PANDORA HAHAHAHAHAHAHA, I will put together a tutorial but in simple words when generating the animations with Pandora the files defaultmale.hkx and defaultfemale.hkx in the folder \meshes\actors\character are generated correctly but other files are not due to the animations of zaz 8+ therefore I generate with all the animations using FNIS those elements are clean, then I generate the pandora in another folder, and I replace that data with what was generated in pandora, good for furniture zap animations and everything the rest. It will work without the problem of nemesis animations. It took me 7 days but I did it. Perform the tests and it works correctly. Tomorrow I will upload a pdf with my observations so that it may be of help so that it can be adopted naturally on the Pandora, a patch for the ZAZ 8+ in which I update the behaviors, and the procedure for Pandora to work. and ZAZ 8+ without major problems. Sorry for the long message. Thanks for the engine, it's fascinating.

is still working? can you share this pdf?

**Comment by anonSkyrim95** [comment]
It's still working, wait a bit because yesterday I became obsessed with creating an .exe that would be able to carry out the entire procedure with a single button, (it's not ready yet) I will upload the PDF with the instructions here, it consists of the starting order of FNIS and Pandora. and then move files from one folder to others manually.

**Comment by AntimonyF5** [comment]
> It's still working, wait a bit because yesterday I became obsessed with creating an .exe that would be able to carry out the entire procedure with a single button, (it's not ready yet) I'll upload the PDF with the instructions here (when I have it ready), it consists of the starting order of FNIS and Pandora. and then move files from one folder to others manually.

Does using the FNIS defaultmale file affect some mods that require Nemesis to work properly?

**Comment by anonSkyrim95** [comment]
I haven't transferred many animation packages to Nemesis and I tested them all after the patch and they work correctly. since if FNIS is executed alone, these animations cannot be executed. wait for the pdf

**Comment by anonSkyrim95** [comment]
[Pandora+ZAZ patch method EN.pdf](https://github.com/Monitor144hz/Pandora-Behaviour-Engine-Plus/files/15154960/Pandora%2BZAZ.patch.method.EN.pdf)
ready the manual, I hope this helps you understand what is happening and why zaz is not entered correctly in pandora, if you follow the manual you will be able to run ZAZA with pandora.

**Comment by bugb2100** [comment]
hello i would like to let you know that by following your method i managed to run ZAZ with Pandora. just want to know that if i install other animation beside ZAZ, do i need to run FNIS first and follow your guide or just run it once?

**Comment by anonSkyrim95** [comment]
Unfortunately, it is necessary to repeat the process for each update. For this reason, I am creating my own executable, which with one or three buttons, first executes FNIS, then Pandora, and finally generates the patch automatically. I hope to have it functional for all systems this week. For now, the one I created only works on my PC. I have to modify some lines so that it is capable of tracking the paths on its own.

**Comment by yunt2049** [comment]
@anonSkyrim95 dude,its work, u r my hero.

**Comment by Monitor144hz** [comment]
this is not a bug, it's just a feature in FNIS that's not present in pandora.

**Comment by Akmitisuhide** [comment]
> this is not a bug, it's just a feature in FNIS that's not present in pandora.

Is there any chance you are working on an official update to fix this issue, because i have a bigmodlist on vortex and dont know if this is possible on it. Of course if its possible to fix it on Vortex i would be glad to know

**Comment by anonSkyrim95** [comment]
![image](https://github.com/Monitor144hz/Pandora-Behaviour-Engine-Plus/assets/167776525/1c65cd5f-60e0-4a7a-83a3-b7b7e23af28e)
The patch is almost ready, it works very well in .pyw "it is just a patch that requires you to install pandora + from the official page, but it is no longer necessary for you to download the FNIS or the XXXXL patches for it." Once the patch is ready I will see how make it work in vortex (I don't promise anything), I hate vortex as I play with almost 4000 mods, vortex destroys the order of the files."

**Comment by Bapios** [comment]
I followed your pdf up until it said to "replace 3 files from Pandora plus export to Pandora+ZAZ8+Fix".

The Files in "\Pandora Plus Export\Meshes" and "\Pandora Plus Export\meshes\actors\Character" were not present for me (Characters, characters female and animationdatasinglefile.txt)

I have opened these folders before the release of pandora when I was trying to make FNIS and Nemisis work together and I don't remember seeing these files in those folders either, I could have missed something in the pdf so I will try it again when I have the time but I'm hoping that the Exe you're working on will fix it.

I have around 500 mods that rely on FNIS and Nemisis. I will also try this on fewer mods to see if one of my mods is causing this issue.

**Comment by Akmitisuhide** [comment]
> ![image](https://private-user-images.githubusercontent.com/167776525/327608513-1c65cd5f-60e0-4a7a-83a3-b7b7e23af28e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ2OTI4NTMsIm5iZiI6MTcxNDY5MjU1MywicGF0aCI6Ii8xNjc3NzY1MjUvMzI3NjA4NTEzLTFjNjVjZDVmLTYwZTAtNGE3YS04M2EzLWI3YjdlMjNhZjI4ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwMlQyMzI5MTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05ZDhjYmM0NjY3OTI5NzNhZjY4MzZhOWNhY2M4ODJmZWIxNjI3ODUyNTdhMmFlMTg3YmU2M2Q4MTNjNDk4MTgzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.wLAiPHeTQzICHjyOqnA1QXMePKFgPt7XbcbQtOwNc1U) The patch is almost ready, it works very well in .pyw "it is just a patch that requires you to install pandora + from the official page, but it is no longer necessary for you to download the FNIS or the XXXXL patches for it." Once the patch is ready I will see how make it work in vortex (I don't promise anything), I hate vortex as I play with almost 4000 mods, vortex destroys the order of the files." I still need pyw to work correctly in exe format.

Thanks in advance, i just used MMM (Mod Manager Migration) to get my hole modlist on MO2, and everything works fine, ill try and follow your tutorial and if nothing works, ill just wait for the patch

**Comment by Bapios** [comment]
Okay so I found 2 mods (SkySA and Ultimate Combat) that were blocking pandora from generating the 3 files (animationdatasinglefile.txt, characters and characters female). After disabling those mods and starting from a new save the animations still won't work.
The only furniture that do work are from Devious Devices. Zaz 8 furniture also didn't work when I had only a few mods enabled.

**Comment by Monitor144hz** [comment]
closing this as a duplicate of #76. Continue conversation there.
